lib/curl_sasl.c

```diff
@@ -661,32 +698,34 @@ CURLcode Curl_sasl_continue(struct SASL *sasl, struct Curl_easy *data,
     sasl->authmechs ^= sasl->authused;
 
     /* Start an alternative SASL authentication */
-    result = Curl_sasl_start(sasl, data, conn, sasl->force_ir, progress);
-    newstate = sasl->state;   /* Use state from Curl_sasl_start() */
-    break;
+    return Curl_sasl_start(sasl, data, conn, sasl->force_ir, progress);
```

Comment:
Is this dropping of the `newstate` assignment on purpose? (just making sure)

Reply:
Yes, because we are returning immediately (`newstate` is a local variable).